src/useTracking.js

```diff
@@ -15,12 +14,12 @@ export default function useTracking() {
 
   return useMemo(
     () => ({
-      getTrackingData: () => trackingContext.tracking.data,
-      trackEvent: data =>
-        trackingContext.tracking.dispatch(
-          merge(trackingContext.tracking.data, data)
-        ),
+      getTrackingData: () => trackingContext.tracking.getTrackingData(),
+      trackEvent: data => trackingContext.tracking.dispatch(data),
```

Comment:
Would it make sense to avoid the extra function declaration?

```js
      getTrackingData: trackingContext.tracking.getTrackingData,
      trackEvent: trackingContext.tracking.dispatch,
````

Reply:
Good call!